batch_job/src/work_queue_factory.c

```diff
@@ -625,7 +696,7 @@ current list of masters, compare it to the number actually submitted, then
 submit more until the desired state is reached.
 */
 
-static void mainloop( struct batch_queue *queue, const char *project_regex, const char *foremen_regex )
```

Comment:
Rather than passing project_regex and foremen_regex to the mainloop (even if we do not use them), I chose to make them global. I can undo this if needed.